overlord/snapstate/handlers_test.go

```diff
@@ -33,21 +34,180 @@ import (
 	"github.com/snapcore/snapd/overlord/state"
 	"github.com/snapcore/snapd/snap"
 	"github.com/snapcore/snapd/testutil"
+	userclient "github.com/snapcore/snapd/usersession/client"
 )
 
 type handlersSuite struct {
 	baseHandlerSuite
 }
 
 var _ = Suite(&handlersSuite{})
+var mockNBR func() = nil
 
 func (s *handlersSuite) SetUpTest(c *C) {
 	s.baseHandlerSuite.SetUpTest(c)
+	if mockNBR == nil {
+		mockNBR = snapstate.MockNotifyBeginRefresh()
+	}
```

Comment:
So this is not really the pattern we usually use, we usually create a mock function in export_test.go and allow mocking of it that way. I'd much prefer you do it like that instead. So in the tests that *need* to mock this, can just call 

```
r := MockNotifyBeginRefresh(....)
defer r()
```

Reply:
The problem it that there are other tests that create a thread that remain after the test has ended, and that thread calls `asyncRefreshOnSnapClose()` after the test has ended, so the function must remain mocked for the whole run.